qcodes/instrument_drivers/tektronix/DPO7200xx.py

```diff
@@ -777,17 +839,37 @@ def _set_source(self, source_number: int, value: str) -> None:
             f"{value}"
         )
 
-    def _measure(self, measurement: str) -> Any:
+    def wait_adjustment_time(self) -> None:
         """
-        Args:
-            measurement: The type of measurement we wish to perform
+        Wait until the minimum time after adjusting the measurement source or
+        type has elapsed
         """
-        if self.type.get_latest() != measurement:
-            self.type(measurement)
-
         time_since_adjust = time.perf_counter() - self._adjustment_time
         if time_since_adjust < self._minimum_adjustment_time:
             time_remaining = self._minimum_adjustment_time - time_since_adjust
             time.sleep(time_remaining)
 
-        return self.ask(f"MEASUrement:MEAS{self._measurement_number}:VALue?")
+
+class TektronixDPOMeasurementStatistics(InstrumentChannel):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        self.add_parameter(
+            "mode",
+            get_cmd="MEASUrement:STATIstics:MODe?",
+            set_cmd="MEASUrement:STATIstics:MODe {}",
+            vals=Enum(
+                "OFF", "ALL", "VALUEMean", "MINMax",
+                "MEANSTDdev"
+            )
+        )
+
+        self.add_parameter(
+            "time_constant",
+            get_cmd="MEASUrement:STATIstics:WEIghting?",
+            set_cmd="MEASUrement:STATIstics:WEIghting {}",
+            get_parser=int
```

Comment:
It's not obvious to me how this works. Would it make sense to include a docstring for this parameter?